addons/project/models/project.py

```diff
@@ -1591,6 +1608,23 @@ def copy_data(self, default=None):
         defaults = super().copy_data(default=default)
         return [{k: v for k, v in default.items() if k in self.SELF_READABLE_FIELDS} for default in defaults]
 
+    @api.model
+    def new(self, values={}, origin=None, ref=None):
+        # Required in case we create a new subtask
+        # as we will update the cache on res.users
+        # and we won't be able to filter the active
+        # users at that time
```

Comment:
That comment really does not help understanding the issue.
How can a portal user try to create a subtask if it doesn't have the access rights to create a task?
And what is "the cache on res.users"?  Is it the value of `self.user_ids` in memory?

Reply:
The portal user can create a task in project sharing feature. In fact, we active the rules for project sharing when at least one collaborator is added in a project (See [here](https://github.com/odoo/odoo/blob/e9134c71b60d20da943978d978b09082f8f4d950/addons/project/models/project_collaborator.py#L41-L57)).

So, if we allow the portal user to see more users that he can (with `portal_user_ids`), when the portal user creates a subtask or a change in the task. The `onchange` method is called by the frontend. In the `onchange`, we will arrive in [`convert_to_record` method](https://github.com/odoo/odoo/blob/b8da632e6e8f008a630a4e43029cd7cbf4bf38a9/odoo/fields.py#L3194) to filter the CoModel records linked to the task, to have only the active ones. So, the `res.users` will be a CoModel in our case, so the system will filter to have the active one but the problem is we have some records (that is some users) that the current user cannot and so we have an `AccessError`.

Yeah the value of `self.user_ids` will always be the users that the current user can see.